sqlalchemy-stubs/orm/session.pyi

```diff
@@ -473,7 +475,7 @@ class Session(_SessionClassMethods):
         _sa_skip_events: Optional[Any] = ...,
         _sa_skip_for_implicit_returning: bool = ...,
     ) -> Union[Connection, Engine]: ...
-    def query(self, *entities: Any, **kwargs: Any) -> Query: ...
+    def query(self, *entities: Any, **kwargs: Any) -> Query[Any]: ...
```

Comment:
should we make an overload here similar to this:
`def query(self, entities: Type[_T], **kwargs: Any) -> Query[_T]`?

not sure if others are also applicable

Reply:
I'm not sure that's right and it's the reason I left `Query.__init__` basically untyped. It uses `coerce.expect()`, so it can be a lot of different things.

Reply:
Without is I don't think we support the case:
`things = session.query(Thing).all()` and things is not `List[Thing]`, no?